gprofiler/docker_client.py

```diff
@@ -0,0 +1,80 @@
+import logging
+import re
+from typing import Dict, Optional, List
+
+import docker
+
+DOCKER_SYSTEMD_CGROUPS = [re.compile(r"/system.slice/docker-([a-z0-9]{64})\.scope")]
+
+logger = logging.getLogger(__name__)
+
+
+class DockerClient:
+    def __init__(self):
+        try:
+            self._client = docker.from_env()
+        except Exception:
+            logger.warning('Could not initiate the Docker client, so the profiling data will not include the container'
+                           ' names. If you are running the gProfiler in a container, please mount the Docker sock file'
+                           'by running the Docker run command with the following argument: '
+                           '"-v /var/run/docker.sock:/var/run/docker.sock". Otherwise, please open a new issue here: '
+                           'https://github.com/Granulate/gprofiler/issues/new')
+            self._client = None
+
+        self._pid_to_container_name_cache: Dict[int, str] = {}
+        self._container_id_to_name_cache: Dict[str, str] = {}
+
+    def reset_cache(self):
+        self._pid_to_container_name_cache = {}
+        self._container_id_to_name_cache = {}
+
+    def get_container_names(self) -> List[str]:
+        return list(self._container_id_to_name_cache.values())
+
+    def get_stack_line_with_container_name(self, stack_line: str, pid: int) -> str:
```

Comment:
The logic of embedding the name into the line repeats multiple times here. Instead, I think a nicer API would be if this function will return the container name or `''` give a PID, and the caller will do the formatting.